source/compiler/tests/unreachable_code.pwn

```diff
@@ -0,0 +1,187 @@
+#include <core>
+#include <console>
+
+new g_var = 0;
+
+test_if_1()
+{
+	if (g_var != 0)
+		return 0;
+	else
+		return 1;
+	return 2; // warning 225: unreachable code
+}
+
+test_if_2()
+{
+	new i = 0;
+	while (i < 10)
+	{
+		if (g_var != 0)
+			return;
+		else
+			continue;
+		i++; // warning 225: unreachable code
+	}
+}
+
+test_if_3()
+{
+	if (g_var != 0)
+		return 0;
+	else
+		return 1;
+	// shouldn't cause warning 209 ("function should return a value")
+}
+
+test_if_4()
+{
+	if (g_var != 0)
+		return 0;
+	else
+		for (;;) {}
+	// shouldn't cause warning 209 ("function should return a value")
+}
+
+test_if_5()
+{
+	if (g_var != 0)
+		return 1;
+	else
+		{}
+	return 1;
+}
+
+test_switch_1()
+{
+	new var0 = 0, var1 = 1, var2 = 2, var3 = 3;
+	switch (g_var)
+	{
+		case 0:
+			return var0;
+		case 1:
+			return var1;
+		default:
+			return var2;
+	}
+	return var3; // warning 225: unreachable code
+}
+
+test_switch_2()
+{
+	new i = 0;
+	while (i < 10)
+	{
+		switch (g_var)
+		{
+			case 0:
+				break;
+			default:
+				continue;
+		}
+		i++; // warning 225: unreachable code
+	}
+	return i;
+}
+
+test_switch_3()
+{
+	switch (g_var)
+	{
+		case 0:
+			return 0;
+		case 1:
+			return 1;
+	}
+	// shouldn't cause warning 225 ("unreachable code")
+	return 2;
+}
+
+test_switch_4()
+{
+	switch (g_var)
+	{
+		case 0:
+			return 0;
+		case 1:
+			return 1;
+		default:
+			return 2;
+	}
+	// shouldn't cause warning 225 ("unreachable code")
+	return 3;
+}
+
+test_endless()
+{
+	if (g_var != 0)
+		return 0;
+	for (;;) {}
+	// shouldn't cause warning 209 ("function should return a value")
```

Comment:
I'm wondering if this should be a different warning.  The code sometimes returns, sometimes doesn't.  The cases where you want a function that doesn't end are extremely rare IMHO, especially compared to the cases where you accidentally wrote an infinite loop and want the compiler to alert you to this fact.

Reply:
Same for any infinite loop anywhere outside `main` TBH.

Reply:
Well, I disabled warning 209 for infinite loops, as usually the compiler warns about them because of the loop condition having a constant result (`warning 206: redundant test: constant expression is non-zero`).
But yes, as you noticed, when a loop condition section of a `for` loop is empty and there are no `break`/`return`/`exit`/`goto` statements inside the loop body, the compiler doesn't warn about the loop being endless.
I've opened a separate issue for this in #587.

@Y-Less, if this is the only problem with this PR, can you approve it for now (and merge it now, if possible)? As I stated in #587. I'll try to add a new warning for infinite loops, but I'll need this PR to be merged first, so later I'll be able to update its tests for the new warning.

Reply:
Yes.  I'll merge things in a few days, as it does take a while to test things, but that's the only questionable thing I've seen.